lib/appFactory.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const path = require('path');
+const { loggingMiddleware } = require('./logger');
+const indexRouter = require('./routes/index');
+const signupRouter = require('./routes/signup');
+const webFingerRouter = require('./routes/webfinger');
+const oAuthRouter = require('./routes/oauth');
+const storageCommon = require('./routes/storage_common');
+const errorPage = require('./util/errorPage');
+const helmet = require('helmet');
+const shorten = require('./util/shorten');
+
+module.exports = function ({ hostIdentity, jwtSecret, account, store, basePath = '' }) {
+  if (basePath && !basePath.startsWith('/')) { basePath = '/' + basePath; }
+
+  const app = express();
+  app.locals.basePath = basePath;
+
+  // view engine setup
+  app.engine('.html', require('ejs').__express);
+  app.engine('.xml', require('ejs').__express);
+  app.set('view engine', 'html');
+  app.set('views', path.join(__dirname, 'views'));
```

Comment:
In my own work I haven't had to build an implementation of express for rendering needs.  I have used it only for API request/response handling where only data is involved.

From what I am parsing here.  If you are trying to tell express to render html and xml files render using the ejs rendering engine ... and for all other file types render using the html rendering engine, then I suppose this will work as you are hoping.

I would recommend just having the ejs rendering engine handle all content being passed here as I think that will just be simpler to implement and hopefully simpler to maintain (`app.set('view engine', 'ejs');`)

Reply:
We actually are having ejs handle all rendering; `app.set('view engine', 'html');` says, when no extension is given, to assume the extension is `.html` (and thus to use EJS).
https://expressjs.com/en/api.html#app.set